extras/dml/data_mover_dml.c

```diff
@@ -91,6 +91,34 @@ data_mover_dml_memmove_job_init(dml_job_t *dml_job,
 	return dml_job;
 }
 
+/*
+ * data_mover_dml_memset_job_init -- initializes new memset dml job
+ */
+static dml_job_t *
+data_mover_dml_memset_job_init(dml_job_t *dml_job,
+	void *ptr, int value, size_t n, uint64_t flags)
+{
+	uint64_t dml_flags = 0;
+	data_mover_dml_translate_flags(flags, &dml_flags);
+
+	dml_job->operation = DML_OP_FILL;
+	dml_job->destination_first_ptr = (uint8_t *)ptr;
+	dml_job->destination_length = n;
+	dml_job->flags = dml_flags;
+
+	/*
+	 * Convert the whole integer 'value' into byte (uint8_t) chunks and
+	 * store them in the 'pattern' array.
+	 */
+	size_t ptrn_idx = 0;
+	while (value != 0) {
+		dml_job->pattern[ptrn_idx] = value & 0xff;
+		value = value >> 8;
+	}
```

Comment:
Could you explain this to me, since I don't fully get it? From what I understand, the value chunks are always stored in `pattern[0]` because the `ptrn_idx` doesn't change. Is that correct?